lib/AstOfLlbc.ml

```diff
@@ -566,47 +680,190 @@ let maybe_addrof (env: env) (ty: C.ty) (e: K.expr) =
   | _ ->
       K.(with_type (TBuf (e.typ, false)) (EAddrOf e))
 
+
+(** Handling trait clauses as dictionaries *)
+
+(* There are two ways that we skip synthesis of trait methods in function calls. The first one is if
+  a trait declaration is blocklisted. This happens if the trait has built-in support (e.g.
+  FnMut), or if the trait relies on unsupported features. The second way we skip trait methods
+  (further down) is if the function is a known builtin implementation. *)
+let blocklisted_trait_decls = [
+  (* Handled primitively. *)
+  "core::ops::function::FnMut";
+  "core::cmp::PartialEq";
+  (* The traits below *should* be handled properly ASAP. But for now, we have specific *instances*
+     of those trait methods in the builtin lookup table, which we then implement by hand with
+     macros. *)
+  "core::iter::traits::iterator::Iterator";
+  "core::iter::range::Step";
+  (* TODO: for now, we leave into as-is in the AST, do a later pass that eliminates all identity
+     calls to into (post-monomorphization), and error our if there are any left that do not operate
+     on primitive types. We should probably remove the special-case in this file and treat it
+     generically with a dedicated pass over the krml ast. *)
+  "core::convert::From";
+  (* TODO: figure out what to do with those *)
+  "core::clone::Clone";
+  "core::fmt::Debug";
+]
+
+(* Using tests/where_clauses_simple as an example.
+
+ fn double<T: Ops + Copy, U: Ops+Copy> (...)
+
+ this gets desugared to fn double<T,U> where
+   T: Ops,      <-- ClauseId 0 (required_methods: add, of_u32)
+   T: Copy,     <-- ClauseId 1 (builtin, so neither required nor provided methods)
```

Comment:
Side note: the important point with `Copy` is that it is a supertrait of `Clone`, so if you implement `Copy` you implement `Clone` (i.e., you have a `clone` method). We may implement a micro-pass in Charon which removes instances of `Copy` whenever possible, and adds a builtin for instances of `Clone` which actually come from `Copy`, so that you don't need to implement any custom support here.

Reply:
that sounds nice!